ArduPlane/GCS_Mavlink.cpp

```diff
@@ -721,6 +721,10 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_do_reposition(const mavlink_com
             requested_position.relative_alt = 0;
         }
 
+        if (!isnan(packet.param3) && packet.param3 > 0) {
+            plane.mode_guided.active_radius_m = constrain_int32(packet.param3, 1, UINT16_MAX);
```

Comment:
I'm wondering if this should be passed as an argument through `set_guided_WP` so that it's clearer. There's also some weird interactions that storing this globally leaves if you are already in guided and then scripting takes over guided, because it's stuck with the radius that had been provided. (The GCS sending a radius will of course make everything else a bit inconsistent, but it's hard to reason about this global state as it currently stands).

Reply:
possibly

Reply:
yup, good idea. I've moved it into set_guided_wp() with a default value of -1 (no change) and where 0 means use the param and else it uses it as-is.